caffe2/CMakeLists.txt

```
@@ -653,6 +657,21 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
       list(APPEND Caffe2_GPU_SRCS
         ${TORCH_SRC_DIR}/csrc/cuda/nccl.cpp)
     endif()
+    if(USE_DISTRIBUTED)
+      if(WIN32)
+        if(BUILD_SPLIT_CUDA)
+          append_filelist("libtorch_cuda_distributed_windows_sources" Caffe2_GPU_SRCS_CPP)
+        else()
+          append_filelist("libtorch_cuda_distributed_windows_sources" Caffe2_GPU_SRCS)
+        endif()
+      else()
+        if(BUILD_SPLIT_CUDA)
+          append_filelist("libtorch_cuda_distributed_sources" Caffe2_GPU_SRCS_CPP)
+        else()
+          append_filelist("libtorch_cuda_distributed_sources" Caffe2_GPU_SRCS)
+        endif()
+      endif()
```

Two suggestions (probably made typo in the code):
 - define filelist target based on `BUILD_SPLIT_CUDA` option
 - Change additive `distributed` into standalone list and add it if not WIN32
```suggestion
        if(BUILD_SPLIT_CUDA)
          set(_target "Caffe2_GPU_SRCS_CPP)
        else()
          set(_target "Caffe2_GPU_SRCS")
        endif()
        append_filelist("libtorch_cuda_base_distributed_sources" ${_target_})
      if(NOT WIN32)
          append_filelist("libtorch_cuda_distributed_c10d_sources" ${_target})
      endif()
```

---

Done.